k_writer/producer.go

```diff
@@ -0,0 +1,81 @@
+package k_writer
+
+import (
+	"github.com/shopify/sarama"
+)
+
+// SimpleProducer publishes Kafka messages. It routes messages to the correct broker, refreshing metadata as appropriate,
+// and parses responses for errors. You must call Close() on a producer to avoid leaks, it may not be garbage-collected automatically when
```

Comment:
Perhaps an example of deferring would be the right thing?

Why doesn't it close and own its own client?